workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectDispatcherSystem.cs

```diff
@@ -11,50 +12,27 @@ namespace Improbable.Gdk.Core.GameObjectRepresentation
     [UpdateInGroup(typeof(SpatialOSReceiveGroup.GameObjectReceiveGroup))]
     internal class GameObjectDispatcherSystem : ComponentSystem
     {
-        private readonly Dictionary<int, SpatialOSBehaviourManager> entityIndexToSpatialOSBehaviourManager =
-            new Dictionary<int, SpatialOSBehaviourManager>();
-        private readonly List<SpatialOSBehaviourManager> spatialOSBehaviourManagers =
-            new Collections.List<SpatialOSBehaviourManager>();
+        private readonly Dictionary<int, MonoBehaviourActivationManager> entityIndexToActivationManager =
+            new Dictionary<int, MonoBehaviourActivationManager>();
+        private readonly Dictionary<int, ReaderWriterStore> entityIndexToReaderWriterStore =
+            new Dictionary<int, ReaderWriterStore>();
 
         public readonly List<GameObjectComponentDispatcherBase> GameObjectComponentDispatchers =
             new List<GameObjectComponentDispatcherBase>();
 
-        internal void AddSpatialOSBehaviourManager(int entityIndex, SpatialOSBehaviourManager spatialOSBehaviourManager)
-        {
-            if (entityIndexToSpatialOSBehaviourManager.ContainsKey(entityIndex))
-            {
-                throw new SpatialOSBehaviourManagerAlreadyExistsException($"SpatialOSBehaviourManager already exists for entityIndex {entityIndex}.");
-            }
+        private RequiredFieldInjector injector;
+        private ILogDispatcher logger;
 
-            entityIndexToSpatialOSBehaviourManager[entityIndex] = spatialOSBehaviourManager;
-            spatialOSBehaviourManagers.Add(spatialOSBehaviourManager);
-        }
-
-        internal void RemoveSpatialOSBehaviourManager(int entityIndex)
+        internal void RemoveActivationManagerAndReaderWriterStore(int entityIndex)
         {
-            if (!entityIndexToSpatialOSBehaviourManager.ContainsKey(entityIndex))
+            if (!entityIndexToActivationManager.ContainsKey(entityIndex))
```

Comment:
Use a TryGet here instead

Reply:
Actually the manager is not used afterwards anymore, I just removed getting the manager afterwards